x/concentrated-liquidity/lp.go

```diff
@@ -107,7 +108,21 @@ func (k Keeper) CreatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 	// Calculate the amount of liquidity that will be added to the pool when this position is created.
 	liquidityDelta := math.GetLiquidityFromAmounts(pool.GetCurrentSqrtPrice(), sqrtPriceLowerTick, sqrtPriceUpperTick, amount0Desired, amount1Desired)
 	if liquidityDelta.IsZero() {
-		return CreatePositionData{}, errors.New("liquidityDelta calculated equals zero")
+
+		// Note that it is impossible to reach the case with both tokens being zero because that case is handled above.
+
+		if !amount0Desired.IsZero() && !amount1Desired.IsZero() {
+			return CreatePositionData{}, fmt.Errorf(`failed to translate amount0 (%d) and amount1 (%d) to positive liquidity. Either try to provide more liquidity or confirm that the desired tick range is still active.
+			If range becomes inactive before getting on chain, you might need to provide more of one token as opposed to two of the original amount`, amount0Desired, amount1Desired)
+		} else if amount0Desired.IsZero() {
+			return CreatePositionData{}, fmt.Errorf(`failed to translate amount1 (%d) to positive liquidity. Either try providing more liquidity or confirm that given range is still inactive.
+			If the given range becomes activated, two tokens will be needed as opposed to one.`, amount1Desired)
+		}
+
+		// amount1Desired is zero
+
+		return CreatePositionData{}, fmt.Errorf(`failed to translate amount0 (%d) to positive liquidity. Either try providing more liquidity or confirm that given range is still inactive.
+			If the given range becomes activated, two tokens will be needed as opposed to one.`, amount0Desired)
```

Comment:
Do you think we should just explicitly check that amount1Desired.IsZero() here and then have an else branch that is "unknown error" for why liquidityDelta is zero? Just might come to bite us later

Reply:
If we construct a boolean table, we do check all logic branches though so just `else` is fine IMO

See this:
https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L68-L70
- true & true

https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L113-L115
- false & false

https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L116-L119
true and false

so `else` must be false and true:
https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L123-L124

Reply:
Either this, or maybe slightly relaxing the language to something like "Possible reasons could be..."